src/Core/Localization/Specification/Factory.php

```diff
@@ -100,7 +100,8 @@ public function buildPriceSpecification(
         CldrLocaleInterface $cldrLocale,
         Currency $currency,
         $numberGroupingUsed,
-        $currencyDisplayType
+        $currencyDisplayType,
+        ?int $maxFractionDigits = null
```

Comment:
You should remove this optional parameter and make it mandatory

Reply:
On second thought, it will be useful if we want to rely on pattern rather than currency values!